broadinstitute/single_cell_portal_core
# Refine cell filtering UI (SCP-5297)

This smoothens a few aspects of the prototype cell filtering UI:

* Fix background color in cell filtering panel
* Enable select / deselect by clicking filter label (not just checkbox)
* Only show loading indicators for cell filtering once button is clicked, like DE loading UX -- this minimizes unnecessary visual distraction, and fixes size issues with the prior loading animation
* Omit user annotations from cell facets (for now)
* Refined user-facing text

It also makes a handful of internal improvements for maintainability, like SCP-consistent function declarations, clearer and less abbreviated names, and more IDE-friendly docstrings.

Here's how it looks.

### Before

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/901e34d2-1bff-482f-ae1f-63caa9b45e9d

### After

https://github.com/broadinstitute/single_cell_portal_core/assets/1334561/7c8d3c84-8257-4800-9d73-0eb48e445b58

Known issues that are not in scope here include:
* Groups can unexpectedly change color upon filtering cluster scatter plot
* Unexpected filtering UI upon changing to non-default annotation
* Not all applicable annotations available in facet dropdown menus

### Test
* Enable filtering via feature flag
* Go to "Human milk - DE pilot" / "Human lactation" study
* Click "Cell filtering" button immediately upon seeing it
* Confirm you can click it, and upon click a loading UI displays in the cell filtering panel
* Wait for filters to load (this is considerably slower in development)
* Confirm panel back button has same color as rest of filtering panel, like DE panel
* Click a filter label
* Confirm that selects the associated checkbox and filters the cluster scatter plot

This relates to SCP-5297.
